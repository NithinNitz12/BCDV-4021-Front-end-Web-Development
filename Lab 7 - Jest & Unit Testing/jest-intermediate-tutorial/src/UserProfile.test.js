import React from "react";
import renderer from "react-test-renderer";
import UserProfile from "./UserProfile";

it("renders correctly when there is a user", () => {
  const user = { name: "John", email: "john@example.com" };
  const tree = renderer.create(<UserProfile user={user} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly when there is no user", () => {
  const tree = renderer.create(<UserProfile />).toJSON();
  expect(tree).toMatchSnapshot();
});
