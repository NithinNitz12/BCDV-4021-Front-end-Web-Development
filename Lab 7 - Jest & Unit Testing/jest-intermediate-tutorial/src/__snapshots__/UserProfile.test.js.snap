// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there is a user 1`] = `
<div>
  <div>
    <h1>
      John
    </h1>
    <p>
      john@example.com
    </p>
  </div>
</div>
`;

exports[`renders correctly when there is no user 1`] = `
<div>
  <p>
    No user information available.
  </p>
</div>
`;
